internal/resourcecache/resourcecache.go

```diff
@@ -0,0 +1,109 @@
+package resourcecache
+
+import (
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var sleepTimeBeforeCleanup = 2 * time.Minute
+
+// ResourceCache is a structure that saves information about a nearly created resource.
+// Resources can be added and retrieved from the cache. A retrieval (ResourceFromCache) also removes the item from the cache.
+// If `sleepTimeBeforeCleanup` has passed, and the resource has not yet been retrieved, ResourceCache considers that resource stale,
+// and removes it.
```

Comment:
By this description it does not look like a cache (as cache hit does not removes the object). Maybe `ResourceStore`?